### DISCUSSION TITLE
Trouble w/ Azure AD B2C (OAUTH_PARSE_PROFILE_ERROR)

### DISCUSSION THREAD

**Discussion by nwo31** [nextauthjs/next-auth#9412] [Help]
I am having an issue with integrating an Azure AD B2C provider in my Next.js application. Here are the fields that I have filled out in my [...nextauth].js file: 

![image](https://github.com/nextauthjs/next-auth/assets/139179451/0b110dab-1af6-4f4b-a632-bdc8ca456689)

I have environment variables set in my .env file where. On the Azure portal, I have created an Azure AD B2C directory where I have registered an application. I have my redirect API set to "http://localhost:3000/api/auth/callback/azure-ad-b2c" and I have also created a user flow as well which supports sign-up and sign-in. ([I followed these steps listed on the documentation](https://next-auth.js.org/providers/azure-ad-b2c)).

I am testing out the user flow of signing up and then signing in, but I am running into an error right after I register for an account. When I click to create a new account, and then register for this account, the account is successfully added to all the users, but an error is thrown and I am redirected to the next-auth login page. The error produced looks like this (I blurred out some of the information): 

![image](https://github.com/nextauthjs/next-auth/assets/139179451/7d80e9ea-220e-47ae-ad48-6357aaa3ff2a)

I am wondering what might be causing this error. I looked online and used ChatGPT, but I have not seen anything like this. Any help would be appreciated and I am willing to also meet virtually.

**Comment by AlexF301** [comment]
Were you able to find out the problem? I'm stuck with this error as well.

**Comment by PrestonSam** [comment]
Bumped into the issue and was able to fix it by debugging the library.

The issue occurs when the OAuth Profile object returned by Azure AD B2C lacks an `emails` key. The default `profile` handler for `AzureADB2CProvider` assumes that such key is present and tries to index it, hence the `reading '0'` error.

Presumably you can modify the scopes that you request so that the B2C endpoint returns an email and prevents this error, but a simpler fix is to insert a custom callback that simply doesn't index for an email, as the argument isn't actually necessary to construct a `User` object.

Here's an example config for constructing `AzureADB2CProvider` without hitting the issue.
Note that the `checks` and `client` bindings are to prevent the error `AADB2C99059`

```ts
    AzureADB2CProvider({
      tenantId: TENANT_ID,
      clientId: CLIENT_ID,
      clientSecret: process.env.AZURE_AD_B2C_CLIENT_SECRET!,
      primaryUserFlow: process.env.AZURE_AD_B2C_PRIMARY_USER_FLOW,
      authorization: {
        params: {
          scope: `offline_access openid`,
        },
      },
      checks: ['pkce'],
      client: {
        token_endpoint_auth_method: 'none',
      },
      profile: (oauthProfile) => ({
        id: oauthProfile.sub,
        name: oauthProfile.name,
      }),
    })
```
